portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/ProfileMgtClientServiceImpl.java

```diff
@@ -166,21 +165,63 @@ public ProfileEntry getProfile(String profileName) throws UserPortalUIException
         }
 
         if (claims.isEmpty()) {
-            return profileEntry.getClaims().stream()
-                    .map(claimConfigEntry -> new ProfileUIEntry(claimConfigEntry, null))
+            return profileEntry.getClaims().stream().map(claimConfigEntry -> new ProfileUIEntry(claimConfigEntry, null))
                     .collect(Collectors.toList());
         }
 
-        return profileEntry.getClaims().stream()
-                .map(claimConfigEntry -> {
-                    Optional<Claim> optional = claims.stream()
-                            .filter(claim -> claim.getClaimUri().equals(claimConfigEntry.getClaimURI()))
-                            .findAny();
-                    if (optional.isPresent()) {
-                        return new ProfileUIEntry(claimConfigEntry, optional.get().getValue());
-                    }
-                    return new ProfileUIEntry(claimConfigEntry, null);
-                }).collect(Collectors.toList());
+        return profileEntry.getClaims().stream().map(claimConfigEntry -> {
+            Optional<Claim> optional = claims.stream()
+                    .filter(claim -> claim.getClaimUri().equals(claimConfigEntry.getClaimURI())).findAny();
+            if (optional.isPresent()) {
+                return new ProfileUIEntry(claimConfigEntry, optional.get().getValue());
+            }
+            return new ProfileUIEntry(claimConfigEntry, null);
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<ProfileUIEntry> getGroupClaimEntries(String uniqueGroupId)
+            throws UserPortalUIException {
+
+        //get all the claims of the group profile as some of the claims may not have been filled yet.
+        ProfileEntry profileEntry = getProfile("group");
+
+        if (profileEntry.getClaims().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<MetaClaim> metaClaims = profileEntry.getClaims().stream()
+                .map(claimConfigEntry -> new MetaClaim(IdentityMgtConstants.CLAIM_ROOT_DIALECT,
+                        claimConfigEntry.getClaimURI())).collect(Collectors.toList());
+
+        List<Claim> claims;
+        try {
+            claims = getRealmService().getIdentityStore().getClaimsOfGroup(uniqueGroupId, metaClaims);
+        } catch (IdentityStoreException e) {
+            LOGGER.error(String.format("Failed to get the user claims for group - %s", uniqueGroupId), e);
+            throw new UserPortalUIException(
+                    String.format("Failed to get the group claims for the profile - group"));
```

Comment:
@GayanM , 
1. Why log error here? 
2. Better to log the information about the meta-claims, if log.error is needed (1)
3. The "Failed to get the user claims for group" is misleading. What is the user here?